core/templates/dev/head/I18nFooter.ts

```diff
@@ -35,34 +35,36 @@ angular.module('oppia').directive('i18nFooter', [
             $http, $timeout, $translate, UserService,
             SUPPORTED_SITE_LANGUAGES) {
           var ctrl = this;
-          // Changes the language of the translations.
-          var preferencesDataUrl = '/preferenceshandler/data';
-          var siteLanguageUrl = '/save_site_language';
-          ctrl.supportedSiteLanguages = SUPPORTED_SITE_LANGUAGES;
+          ctrl.$onInit = function() {
+            // Changes the language of the translations.
+            var preferencesDataUrl = '/preferenceshandler/data';
+            var siteLanguageUrl = '/save_site_language';
+            ctrl.supportedSiteLanguages = SUPPORTED_SITE_LANGUAGES;
 
-          // The $timeout seems to be necessary for the dropdown to show
-          // anything at the outset, if the default language is not English.
-          $timeout(function() {
-            // $translate.use() returns undefined until the language file is
-            // fully loaded, which causes a blank field in the dropdown, hence
-            // we use $translate.proposedLanguage() as suggested in
-            // http://stackoverflow.com/a/28903658
-            ctrl.currentLanguageCode = $translate.use() ||
-              $translate.proposedLanguage();
-          }, 50);
+            // The $timeout seems to be necessary for the dropdown
+            // to show anything at the outset, if the default language
+            // is not English.
+            $timeout(function() {
+              // $translate.use() returns undefined until the language
+              // file is fully loaded, which causes a blank field
+              // in the dropdown, hence we use $translate.proposedLanguage()
+              // as suggested in http://stackoverflow.com/a/28903658
+              ctrl.currentLanguageCode = $translate.use() ||
+                $translate.proposedLanguage();
+            }, 50);
 
-          ctrl.changeLanguage = function() {
-            $translate.use(ctrl.currentLanguageCode);
-            UserService.getUserInfoAsync().then(function(userInfo) {
-              if (userInfo.isLoggedIn()) {
-                $http.put(siteLanguageUrl, {
-                  site_language_code: ctrl.currentLanguageCode
-                });
-              }
-            });
+            ctrl.changeLanguage = function() {
```

Comment:
Here and elsewhere, I don't think the functions should be defined in onInit, right? Only the parts that should run when the controller/directive/etc. is initialized should be.

See the example for editableField.js in https://docs.angularjs.org/guide/component ("Example of a component tree").

This probably needs to be fixed elsewhere too. (One good thing, I suppose, is that it will reduce the diff.)

Reply:
Done.

Reply:
Updated.